src/drivers/uavcan_v1/Subscribers/Battery.hpp

```diff
@@ -41,43 +41,149 @@
-	};
+			bat_status.connected = true; // Based on some threshold or an error??
+
+			// Estimate discharged mah from Joule
+			if (_nominal_voltage != NAN) {
+				bat_status.discharged_mah = (source_ts.value.full_energy.joule - source_ts.value.energy.joule)
+							    / (_nominal_voltage * WH_TO_JOULE);
+			}
+
+			bat_status.remaining = source_ts.value.energy.joule / source_ts.value.full_energy.joule;
+
+			// TODO uORB publication rate limiting
+			_battery_status_pub.publish(bat_status);
+			print_message(bat_status);
+
+
+		} else if (receive.port_id == _status_sub._canard_sub._port_id) {
+			reg_drone_service_battery_Status_0_2 bat {};
+			size_t bat_size_in_bytes = receive.payload_size;
+			reg_drone_service_battery_Status_0_2_deserialize_(&bat, (const uint8_t *)receive.payload, &bat_size_in_bytes);
+
+			bat_status.scale = -1; // What does the mean?
+
+			bat_status.temperature = bat.temperature_min_max[1].kelvin - KELVIN_OFFSET; // PX4 uses degC we assume
+
+			bat_status.cell_count = bat.cell_voltages.count;
+
+			uint32_t cell_count = bat_status.cell_count;
+
+			if (cell_count > (sizeof(bat_status.voltage_cell_v) / sizeof(bat_status.voltage_cell_v[0]))) {
+				cell_count = sizeof(bat_status.voltage_cell_v) / sizeof(bat_status.voltage_cell_v[0]);
+			}
+
+			float voltage_cell_min = FLT_MAX_EXP;
+			float voltage_cell_max = FLT_MIN_EXP;
+
+			for (uint32_t i = 0; i < cell_count; i++) {
+				bat_status.voltage_cell_v[i] = bat.cell_voltages.elements[i];
+
+				if (bat_status.voltage_cell_v[i] > voltage_cell_max) {
+					voltage_cell_max = bat_status.voltage_cell_v[i];
+				}
+
+				if (bat_status.voltage_cell_v[i] < voltage_cell_min) {
+					voltage_cell_min = bat_status.voltage_cell_v[i];
+				}
+			}
+
+			bat_status.max_cell_voltage_delta = voltage_cell_max - voltage_cell_min; // Current delta or max delta over time?
+
+		} else if (receive.port_id == _parameters_sub._canard_sub._port_id) {
+			reg_drone_service_battery_Parameters_0_3 parameters {};
+			size_t parameters_size_in_bytes = receive.payload_size;
+			reg_drone_service_battery_Parameters_0_3_deserialize_(&parameters,
+					(const uint8_t *)receive.payload,
+					&parameters_size_in_bytes);
+
+			bat_status.capacity = parameters.design_capacity.coulomb / 3.6f; // Coulomb -> mAh
+			bat_status.cycle_count = parameters.cycle_count;
+			bat_status.serial_number = parameters.unique_id & 0xFFFF;
+			bat_status.state_of_health = parameters.state_of_health_pct;
+			bat_status.max_error = 1; // UAVCAN didn't spec'ed this, but we're optimistic
+			bat_status.id = 0; //TODO instancing
+			_nominal_voltage = parameters.nominal_voltage.volt;
+		}
+	}
+
+private:
+	float _nominal_voltage = NAN;
+
+	uORB::PublicationMulti<battery_status_s> _battery_status_pub{ORB_ID(battery_status)};
+
+	SubjectSubscription _status_sub;
+	SubjectSubscription _parameters_sub;
+
+	const char *_status_name = "battery_status";
+	const char *_parameters_name = "battery_parameters";
```

Comment:
Same as I wrote above --- did you mean [`battery.status`, `battery.parameters`](https://github.com/UAVCAN/public_regulated_data_types/tree/master/reg/drone#port-naming-conventions)? This is a mere convention but it costs nothing to uphold.

Reply:
I took it over from https://github.com/UAVCAN/public_regulated_data_types/blob/master/reg/drone/service/battery/_.0.1.uavcan#L4

Resulting in "uavcan.sub.battery_status.0.id" not sure if this right didn't use or test the param system yet.

Reply:
> I took it over from https://github.com/UAVCAN/public_regulated_data_types/blob/master/reg/drone/service/battery/_.0.1.uavcan#L4

My bad, sorry. You are right.